git/config.go

```diff
@@ -239,5 +243,68 @@ func ParseConfig(configFile io.Reader) GitConfig {
 			sections = append(sections, *section)
 		}
 	}
-	return GitConfig{sections}
+	return GitConfig{sections: sections}
+}
+
+func LoadGlobalConfig() (GitConfig, error) {
+	home := os.Getenv("HOME")
+	if home == "" {
+		home = os.Getenv("home") // On some OSes, it is home
+	}
+
+	if home == "" {
+		return GitConfig{}, fmt.Errorf("Global git configuration could not be found since HOME and home environment variables were not defined.")
+	}
+
+	fname := home + "/.gitconfig"
```

Comment:
I think everything above this should be replaced with a private function to get the global config which can be modified with build tags for different OSes, then we could use,  for instance $home/lib/gitconfig where it sense. (Also, I suspect it's only a matter of time before someone asks for xdg config support)

Reply:
👍